toolkits/dropbox/arcade_dropbox/utils.py

```diff
@@ -0,0 +1,106 @@
+import json
+from typing import Any, Optional
+
+import httpx
+
+from arcade_dropbox.constants import (
+    API_BASE_URL,
+    API_VERSION,
+    ENDPOINT_URL_MAP,
+    Endpoint,
+    EndpointType,
+)
+from arcade_dropbox.exceptions import DropboxApiError
+
+
+def build_dropbox_url(endpoint_type: EndpointType, endpoint_path: str) -> str:
+    base_url = API_BASE_URL.format(endpoint_type=endpoint_type.value)
+    return f"{base_url}/{API_VERSION}/{endpoint_path.strip('/')}"
+
+
+def build_dropbox_headers(token: Optional[str]) -> dict[str, str]:
+    return {"Authorization": f"Bearer {token}"} if token else {}
+
+
+def build_dropbox_json(**kwargs: Any) -> dict:
+    return {key: value for key, value in kwargs.items() if value is not None}
+
+
+async def send_dropbox_request(
+    authorization_token: Optional[str],
+    endpoint: Endpoint,
+    **kwargs: Any,
+) -> Any:
+    endpoint_type, endpoint_path = ENDPOINT_URL_MAP[endpoint]
+    url = build_dropbox_url(endpoint_type, endpoint_path)
+    headers = build_dropbox_headers(authorization_token)
+    json_data = build_dropbox_json(**kwargs)
+
+    if json_data.get("cursor"):
+        url += "/continue"
+        # If cursor is provided, every other argument must be ignored to avoid API error
+        json_data = {"cursor": json_data["cursor"]}
+
+    if endpoint_type == EndpointType.CONTENT:
+        headers["Dropbox-API-Arg"] = json.dumps(json_data)
+        json_data = {}
+
+    async with httpx.AsyncClient() as client:
```

Comment:
why not use the [Offical Dropbox Python SDK](https://github.com/dropbox/dropbox-sdk-python)?

Reply:
it should probably work fine on newer Python versions. but the main issue I think is that all HTTP requests are blocking. using httpx will provide more freedom in future tools that require more complex networking patterns without compromising performance